AirCasting/Map/GoogleMapView.swift

```diff
@@ -166,8 +161,9 @@ struct GoogleMapView: UIViewRepresentable {
     }
     
     func color(point: PathPoint) -> UIColor {
-        guard let thresholds = threshold, let selectedStream = selectedStream else { return .white }
-        let measurement = selectedStream.isTemperature && userSettings.convertToCelsius ? Int32(TemperatureConverter.calculateFahrenheit(celsius: point.measurement)) : Int32(point.measurement)
+        guard let thresholds = threshold else { return .white }
+        let formatter = ThresholdFormatter(for: thresholds)
+        let measurement = formatter.formattedToFahrenheit(for: point)
```

Comment:
good stuff, but the caller code should not decide to what it shall be formatting (the `Fahrenheit` part here).
Perfect-ish interface would be something like that:
```Swift
protocol ThresholdFormatter {
    func formattedValue(for: SensorThreshold) -> String
    func formattedUnit(for: SensorThreshold) -> String
}
```

Which will completely hide the fact that we're formatting temperature units under the hood, the view would only do something like:

```Swift
let formatter = Resolver.resolve(ThresholdFormatter.self, args: stream) //stream is of MeasurementStream type
let value = formatter.formattedValue(for: threshold)
let unit = formatter.formattedUnit(for: threshold)
```

See how now we have all formatting under control in a single place, if we wanted to add Kelvins next, we would only modify the formatter file/class without the need to visit all the places like here when we explicitly ask for "hey, format this value to fahrenheit". Instead we do it like "hey, give me a proper formatter for this stream" which is hiding all this.

Reply:
Thank you for your input, but honestly, I cannot find a way to solve this as you mentioned. 

The only thing that comes to my mind is to change function names to hide the fact that we format our values to exact temperature but I cannot find a way to change function parameters as you advised using a just `threshold`. Every function used a different parameter type like `Path Point`, `Double`, or `String`. 

Now If we want to add a new unit like Kelvins we will have to change only our file with formatter. - maybe I am wrong